academy/ibc/ibc-tao-dev.md

```diff
@@ -0,0 +1,314 @@
+
+Also you can see how it creates a new `ConnectionEnd`:
+
+```go
+connection := types.NewConnectionEnd(types.INIT, clientID, counterparty, types.ExportedVersionsToProto(versions), delayPeriod)
+````
+
+`ConnOpenInit` is triggered by the **relayer**, which constructs the message and hands it over to the SDK that uses the [`msg_server.go`](https://github.com/cosmos/ibc-go/blob/main/modules/core/keeper/msg_server.go) you saw before to call `ConnOpenInit`:
+
+```go
+
+// ConnectionOpenInit defines a rpc handler method for MsgConnectionOpenInit.
+func (k Keeper) ConnectionOpenInit(goCtx context.Context, msg *connectiontypes.MsgConnectionOpenInit) (*connectiontypes.MsgConnectionOpenInitResponse, error) {
+  ctx := sdk.UnwrapSDKContext(goCtx)
+
+  if _, err := k.ConnectionKeeper.ConnOpenInit(ctx, msg.ClientId, msg.Counterparty, msg.Version, msg.DelayPeriod); err != nil {
+    return nil, sdkerrors.Wrap(err, "connection handshake open init failed")
+  }
+
+  return &connectiontypes.MsgConnectionOpenInitResponse{}, nil
+}
+```
+
+<HighlightBox type="info">
+
+In IBC, blockchains do not directly pass messages to each other over the network. To communicate, blockchains commit the state to a precisely defined path reserved for a specific message type and a specific counterparty. In the case of establishing a connection,  Relayers monitor for updates on these paths and relay messages by submitting the data stored under the path along with proof of that data to the counterparty chain.
+
+The paths that all IBC implementations must support for committing IBC messages are defined in the [ICS-24 host state machine requirements](https://github.com/cosmos/ics/tree/master/spec/core/ics-024-host-requirements). The proof format that all implementations must produce and verify is defined in the [ICS-23 implementation documentation from Confio](https://github.com/confio/ics23).
+
+</HighlightBox>
+
+**OpenTry**
+
+This `OpenInit` will be followed by an `OpenTry` response. in which chain B will verify the identity of chain A according to information that chain B has about chain A in its light client (the algorithm and the last snapshot of the consensus state containing the root hash of the latest height as well as the next validator set), as well as respond to some of the information about its own identity in the `OpenInit` announcement from chain A. 
+
+![OpenTry](/academy/ibc/images/open_try.png)
+
+The purpose of this handshake is not only for chain B to verify that the counterparty chain A is indeed the expected counterparty identity, but also to verify that the counterparty has the accurate information about chain B's identity. The relayer will also submit two `UpdateClient`s with chain A and chain B as source chains before this handshake. These `UpdateClient` will update both chain A and chain B light clients, in order to make sure that the state verifications in this step are successful.
```

Comment:
Add term 'double verification' + "The purpose of this step of the handshake..." instead of "The purpose of this handshake'